Blish HUD/GameServices/Modules/Pkgs/GitHubPkgRepoProvider.cs

```diff
@@ -0,0 +1,127 @@
+                var manifests = await LoadPkgManifests(releaseResult.Releases);
+
+                if (manifests.Exception == null) {
+                    return manifests.PkgManifests;
+                } else {
+                    progress?.Report($"{Strings.GameServices.ModulesService.PkgManagement_Progress_FailedToReadOrParseRepoManifest}\r\n{manifests.Exception.Message}");
+                }
+            } else {
+                progress?.Report($"{Strings.GameServices.ModulesService.PkgManagement_Progress_FailedToGetReleases}\r\n{releaseResult.Exception.Message}");
+            }
+
+            return null;
+        }
+
+        private async Task<(GitHubRelease[] Releases, Exception Exception)> UpdateGitHubReleases() {
+            try {
+                return (await $"{this.RepoUrl}{GITHUB_RELEASES_URI}".WithHeader("User-Agent", "Blish-HUD").GetJsonAsync<GitHubRelease[]>(), null);
+            } catch (FlurlHttpException ex) {
+                return (new GitHubRelease[0], ex);
+            }
+        }
+
+        private async Task<(PkgManifest[] PkgManifests, Exception Exception)> LoadPkgManifests(IEnumerable<GitHubRelease> releases) {
+            Exception lastException = null;
+
+            foreach (var release in releases) {
+                string compressedReleaseUrl = release.Assets.First(asset => asset.Name.Equals(ASSET_PACKAGE_NAME, StringComparison.InvariantCultureIgnoreCase)).BrowserDownloadUrl;
+
+                try {
+                    using var compressedRelease = await compressedReleaseUrl.GetStreamAsync();
+                    compressedRelease.Seek(-4, SeekOrigin.End);
+                    var lengthBytes = new byte[4];
+                    await compressedRelease.ReadAsync(lengthBytes, 0, 4);
+
+                    int length = BitConverter.ToInt32(lengthBytes, 0);
+                    compressedRelease.Position = 0;
+                    using var gzipStream = new GZipStream(compressedRelease, CompressionMode.Decompress);
+                    var       result     = new byte[length];
+                    gzipStream.Read(result, 0, length);
+
+                    string rawJson = Encoding.UTF8.GetString(result);
+                    return (JsonConvert.DeserializeObject<PkgManifest[]>(rawJson), null);
+                } catch (Exception ex) {
+                    Logger.Warn(ex, $"Failed to load release list from '{compressedReleaseUrl}'.");
+                    lastException = ex;
+                }
+            }
+
+            return (null, lastException);
+        }
+
+        public IEnumerable<PkgManifest> GetPkgManifests(params Func<PkgManifest, bool>[] filters) {
+            foreach (var pkg in from pkg in _pkgCache[this.RepoUrl]
+                                let include = filters.All(filter => filter(pkg))
+                                where include
+                                select pkg) {
+                yield return pkg;
+            }
```

Comment:
IMO, the query syntax is harder to read than the function syntax, even though the function syntax just feels like boilerplate when writing the code.
The whole function could be rewritten as:
```cs
        public IEnumerable<PkgManifest> GetPkgManifests(params Func<PkgManifest, bool>[] filters) {
            return _pkgCache[this.RepoUrl].Where(pkg => filters.All(filter => filter(pkg)));
        }
```
I'm pretty sure you could even change the `IEnumerable` to `IQueryable` without any problem

Reply:
I didn't actually mean to leave this query syntax in.  I find it quite more difficult to read, too.  ReSharper must have been getting antsy.  😁